jmx-scraper/README.md

```diff
@@ -21,20 +21,70 @@ Minimal configuration required
 Configuration can be provided through:
 
 - command line arguments:
-  `java -jar scraper.jar --config otel.jmx.service.url=service:jmx:rmi:///jndi/rmi://tomcat:9010/jmxrmi otel.jmx.target.system=tomcat`.
+  `java -jar scraper.jar -config otel.jmx.service.url=service:jmx:rmi:///jndi/rmi://tomcat:9010/jmxrmi otel.jmx.target.system=tomcat`.
 - command line arguments JVM system properties:
   `java -Dotel.jmx.service.url=service:jmx:rmi:///jndi/rmi://tomcat:9010/jmxrmi -Dotel.jmx.target.system=tomcat -jar scraper.jar`.
 - java properties file: `java -jar scraper.jar -config config.properties`.
 - stdin: `java -jar scraper.jar -config -` where `otel.jmx.target.system=tomcat` and
   `otel.jmx.service.url=service:jmx:rmi:///jndi/rmi://tomcat:9010/jmxrmi` is written to stdin.
+- environment variables: `OTEL_JMX_TARGET_SYSTEM=tomcat OTEL_JMX_SERVICE_URL=service:jmx:rmi:///jndi/rmi://tomcat:9010/jmxrmi java -jar scraper.jar`
 
-TODO: update this once autoconfiguration is supported
+SDK auto-configuration is being used, so all the configuration options can be set using the java
+properties syntax or the corresponding environment variables.
 
-### Configuration reference
+For example the `otel.jmx.service.url` option can be set with the `OTEL_JMX_SERVICE_URL` environment variable.
 
-TODO
+## Configuration reference
 
-### Extra libraries in classpath
+| config option                     | description                                                                                  |
+|-----------------------------------|----------------------------------------------------------------------------------------------|
+| `otel.jmx.service.url`            | mandatory JMX URL to connect to the remote JVM                                               |
+| `otel.jmx.target.system`          | comma-separated list of systems to monitor, mandatory unless a custom configuration is used  |
+| `otel.jmx.custom.scraping.config` | path to a custom YAML metrics definition, mandatory when `otel.jmx.target.system` is not set |
+| `otel.jmx.username`               | user name for JMX connection                                                                 |
+| `otel.jmx.password`               | password for JMX connection                                                                  |
```

Comment:
Doesn't have to be in this PR of course, but it might be nice to have an indication of which configs are required/optional and/or what the default values are.

Reply:
I tried to do that with dedicated column but this wasn't looking great because
- there is only a single mandatory option for the remote URL
- other arguments are mandatory depending on the context (authentication enabled on remote JVM, if there is already a target system set or a custom yaml file used).

This is why I decided to document this mostly in the config option descriptions, I've clarified that for jmx authentication in d237a56